kayenta-core/src/main/java/com/netflix/kayenta/canary/providers/metrics/QueryConfigUtils.java

```diff
@@ -147,7 +147,11 @@ private static void populateTemplateBindings(
           }
         }
 
-        String propertyValue = (String) propertyDescriptor.get().getReadMethod().invoke(bean);
+        // Some stuff like start, end and step can't be cast to a string.
+        String propertyValue =
```

Comment:
@duftler this is the only change that could effect other integrations, I think it is safe though.

Reply:
This change seems safe to me as well.